### DISCUSSION TITLE
ESBMC buidling issue

### DISCUSSION THREAD

**Discussion by ghost** [esbmc/esbmc#1346] [General]
Getting an error while building ESBMC on WSL (during the final cmake step, just before the final make; all previous steps described in BUILDING.md apparently worked succesfully). Any hint what could be going wrong here:

myHome@myHome-mobl2:~/tools/ESBMC_compile/ESBMC_Project/esbmc/build$ cmake .. -GNinja -DBUILD_TESTING=On -DENABLE_REGRESSION=On -DClang_DIR=$PWD/../../clang13 -DLLVM_DIR=$PWD/../../clang13 -DBUILD_STATIC=On -DBoolector_DIR=$PWD/../../boolector-release -DZ3_DIR=$PWD/../../z3 -DENABLE_MATHSAT=ON -DMathsat_DIR=$PWD/../../mathsat -DENABLE_YICES=On -DYices_DIR=$PWD/../../yices -DCVC4_DIR=$PWD/../../cvc4 -DGMP_DIR=$PWD/../../gmp -DBitwuzla_DIR=$PWD/../../bitwuzla-release -DCMAKE_INSTALL_PREFIX:PATH=$PWD/../../release
-- Detected non-apple Unix
-- LIBGOMP_LIB: -lgomp -ldl
-- OS_FLEX_SMTLIB_FLAGS:
-- OS_X86_INCLUDE_FOLDER: /usr/include/x86_64-linux-gnu
-- OS_C2GOTO_FLAGS:
-- OS_Z3_LIBS: stdc++ -pthread
-- sendfile found, esbmc_parseoptions will use it
-- ESBMC will be built in static mode
Looking for LLVM in: /home/myHome/tools/ESBMC_compile/ESBMC_Project/esbmc/build/../../clang13
-- Clang is CHERI-enabled, enabling CHERI support.
-- LLVM version: 13.0.0git
-- Clang found in: /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13
-- Linking libclang: static
**CMake Error at scripts/cmake/FindLocalLLVM.cmake:71 (message):
  Cannot compile against Clang: Change Dir:
  '/home/myHome/tools/ESBMC_compile/ESBMC_Project/esbmc/build/CMakeFiles/CMakeTmp'**

  Run Build Command(s): /usr/bin/ninja -v cmTC_43276

  [1/2] /usr/bin/c++
  -I/home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/include -static
  -fPIC -std=gnu++17 -fPIE -o CMakeFiles/cmTC_43276.dir/try_clang.cc.o -c
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/esbmc/scripts/cmake/try_clang.cc

  [2/2] : && /usr/bin/c++ -static -fPIC -static
  CMakeFiles/cmTC_43276.dir/try_clang.cc.o -o cmTC_43276
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangTooling.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangAST.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangIndex.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangFormat.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangToolingInclusions.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangFrontend.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangDriver.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMOption.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangParse.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangSerialization.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangSema.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangEdit.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangAnalysis.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangASTMatchers.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangAST.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMFrontendOpenMP.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMTransformUtils.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMAnalysis.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMProfileData.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMObject.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMMCParser.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMTextAPI.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMBitReader.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangToolingCore.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangRewrite.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangLex.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangBasic.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMMC.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMDebugInfoCodeView.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMCore.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMBinaryFormat.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMRemarks.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMBitstreamReader.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMSupport.a
  -lrt -ldl -lpthread -lm /usr/lib/x86_64-linux-gnu/libz.a
  /usr/lib/x86_64-linux-gnu/libtinfo.so
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMDemangle.a
  && :

  FAILED: cmTC_43276

  : && /usr/bin/c++ -static -fPIC -static
  CMakeFiles/cmTC_43276.dir/try_clang.cc.o -o cmTC_43276
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangTooling.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangAST.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangIndex.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangFormat.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangToolingInclusions.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangFrontend.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangDriver.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMOption.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangParse.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangSerialization.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangSema.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangEdit.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangAnalysis.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangASTMatchers.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangAST.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMFrontendOpenMP.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMTransformUtils.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMAnalysis.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMProfileData.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMObject.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMMCParser.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMTextAPI.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMBitReader.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangToolingCore.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangRewrite.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangLex.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libclangBasic.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMMC.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMDebugInfoCodeView.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMCore.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMBinaryFormat.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMRemarks.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMBitstreamReader.a
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMSupport.a
  -lrt -ldl -lpthread -lm /usr/lib/x86_64-linux-gnu/libz.a
  /usr/lib/x86_64-linux-gnu/libtinfo.so
  /home/myHome/tools/ESBMC_compile/ESBMC_Project/clang13/lib/libLLVMDemangle.a
  && :

  /usr/bin/ld: attempted static link of dynamic object
  `/usr/lib/x86_64-linux-gnu/libtinfo.so'

  collect2: error: ld returned 1 exit status

  ninja: build stopped: subcommand failed.

Call Stack (most recent call first):
  CMakeLists.txt:91 (include)

-- Configuring incomplete, errors occurred!

**Comment by fbrausse** [comment]
> /usr/bin/ld: attempted static link of dynamic object
`/usr/lib/x86_64-linux-gnu/libtinfo.so'

libtinfo is a library from ncurses. Can you check whether there is `/usr/lib/x86_64-linux-gnu/libtinfo.a` (note: .a instead of .so) installed on your system?

**Reply by ghost** [reply]
Yes, I do see the file libtinfo.a at the location /usr/lib/x86_64-linux-gnu/libtinfo.a. Do I need to make any change in any script?

**Reply by fbrausse** [reply]
> Do I need to make any change in any script?

I don't know, yet.

I've opened an issue from this discussion page. Let's track the problem down there: https://github.com/esbmc/esbmc/issues/1359
